src/ngraph/op/fused/normalize_l2.cpp

```diff
@@ -42,65 +42,66 @@ op::NormalizeL2::NormalizeL2(const Output<Node>& data,
 
 void op::NormalizeL2::pre_validate_and_infer_types()
 {
-    const auto& data_pshape = get_input_partial_shape(0);
+    auto axes_node = input(1).get_source_output().get_node_shared_ptr();
+    const auto& input_pshape = get_input_partial_shape(0);
     const auto& axes_pshape = get_input_partial_shape(1);
+    const auto& input_rank = input_pshape.rank();
+    const auto& axes_rank = axes_pshape.rank();
 
-    NODE_VALIDATION_CHECK(this, data_pshape.is_static(), "Input data must be static.");
-    NODE_VALIDATION_CHECK(this, axes_pshape.is_static(), "Input axes must be static.");
-
-    const Shape data_shape{data_pshape.to_shape()};
-
-    // Input data must be 2, 3 or 4D tensor.
     NODE_VALIDATION_CHECK(this,
-                          (data_shape.size() >= 2 && data_shape.size() <= 4),
-                          "Input tensor rank must be 2, 3 or 4 dimensional (actual input "
-                          "shape: ",
-                          data_shape,
-                          ").");
+                          axes_node->is_constant(),
+                          "doesn't support 'axes' input of other type than a Constant.");
 
-    NODE_VALIDATION_CHECK(this,
-                          static_cast<size_t>(axes_pshape.rank()) == 1,
-                          "Input axes must have rank equals 1 (axes shape: ",
-                          axes_pshape,
-                          ").");
+    if (axes_rank.is_static())
+    {
+        NODE_VALIDATION_CHECK(this,
+                              static_cast<size_t>(axes_pshape.rank()) == 1,
```

Comment:
Perhaps you could use `axes_rank` instead of `axes_pshape.rank()` just for consistency

Reply:
Right, done